(/customization/user-button) Add Vue examples; update copy

wobsoriano opened the pull request:
### 🔎 Previews:

- https://clerk.com/docs/pr/1999/customization/user-button

### What does this solve?

- Since the implementation uses Vue [slots](https://vuejs.org/guide/components/slots) (rather than props like in React) to pass in custom components, it's important to document this approach.

### What changed?

- Adds Vue examples to each section of the custom menu items docs
- Shows proper usage of template slots for custom components

### Checklist

- [x] I have clicked on "Files changed" and performed a thorough self-review
- [x] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [x] All existing checks pass

alexisintech reviewed (approved):
awesome, I've updated the copy on this page as it was much needed, and thus, updated the PR title.
all examples seem to be working for me locally.
thanks so much for being on top of this! would you say that these examples would pretty much be the same for nuxt?

(I've approved, feel free to merge when you're ready)

wobsoriano commented:
> awesome, I've updated the copy on this page as it was much needed, and thus, updated the PR title. all examples seem to be working for me locally. thanks so much for being on top of this! would you say that these examples would pretty much be the same for nuxt?
> 
> (I've approved, feel free to merge when you're ready)

Yes same for Nuxt!
